src/fairness/writer/data_writer_base.hpp

```diff
@@ -0,0 +1,37 @@
+/************************************************************\
+ * Copyright 2021 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+using namespace Flux::accounting;
+
+namespace Flux {
+namespace writer {
+
+/*!  Base data writer class.
+*/
+class data_writer_base_t {
+public:
+    virtual ~data_writer_base_t () = default;
+
+    int write_accounting_info (const std::string &path);
```

Comment:
Is this function ever implemented?  Maybe it is worth refactoring this so that the db writer subclass has a function with the same name/arguments (i.e., this function should grow a `root` argument and the db subclass should drop the `db` part of its method's name).

Reply:
Might also be instructive to think if you could implement the `csv` writer and the `stdout` writers in terms of this base class.  In which case, one option might be to move some arguments (e.g., db/csv path) into the constructor.

Reply:
I hope I understood this correctly; apologies if I didn't! I looked at flux-sched's [resource readers class](https://github.com/flux-framework/flux-sched/tree/master/resource/readers) for some clarification. 

For my own understanding, is the concept here that sub-classes **override** the base class's functions when a subclass object is created and its member functions are called?

Reply:
> For my own understanding, is the concept here that sub-classes override the base class's functions when a subclass object is created and its member functions are called?

Yeah, exactly.  Check out [this wiki article](https://en.wikipedia.org/wiki/Method_overriding) for more info.

Reply:
Thanks @cmoussa1! The changes you made are exactly what I recommended in my first comment.  So I think this PR is good to go and I can resolve this comment.  As a future refactor (certainly doesn't need to be done in this PR), it would be cool if the `flux shares` stdout output writer could be re-written as a subclass of `data_writer_base_t`, which would probably require some refactoring of the `write_acct_info` method's arguments (that's what I was poorly hinting at in my second comment).